resources/smcp-templates/v2.3/maistra

```diff
@@ -29,7 +29,7 @@ spec:
         container:
           imageRegistry: quay.io/openshift
           imageName: origin-oauth-proxy
-          imageTag: "4.4"
+          imageTag: "4.9"
```

Comment:
I'm afraid this doesn't do what you want.

I think you will want to change this
https://github.com/maistra/istio-operator/blob/maistra-2.3/deploy/src/deployment-servicemesh.yaml#L53

However I'm not sure if we should bump this in a z release or if it's better to do this in 2.4.

Reply:
For prometheus and grafana, I found in their respective deployments ([prometheus](https://github.com/maistra/istio-operator/blob/fbc41861f7569f05fec20cd21247983287fd21c7/resources/helm/overlays/istio-telemetry/prometheus/templates/deployment.yaml#L40), [grafana](https://github.com/maistra/istio-operator/blob/81c86c46f8b195f20be7a0a1ada207f1a48bf587/resources/helm/overlays/istio-telemetry/grafana/templates/deployment.yaml#L48)) that the version of oauth-proxy that they reference in their deployments depend on this `global.oauthproxy` variable which I found defined in that smcp-templates file, so I changed that one. 

This change does affect the versions of oauth-proxy used in prometheus-proxy and grafana-proxy, and had the desired effect in my testing, as before I made that change those proxies were unable to work with bcrypt.

Would updating the imageTag in the file you referenced maybe update a different oauth proxy in our deployment of the operator? I'm unsure.

Reply:
I'm concerned with your comment: `Update the oauth-proxy version from 4.4 -> 4.9 (first version to support bcrypt)`

The link I posted is very likely the source of truth. The file you updated in this PR is mostly unused (they are only used if the customer makes use of templates, which by default they don't).

So, if bcrypt is only supported on oauth-proxy 4.9 and by default we use 4.4, then this PR will break things.

Reply:
I'm putting a hold on this PR until someone more experienced in operator can confirm this.
cc @rcernich @luksa

Reply:
We need to re-target this PR for 2.4.